spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java

```diff
@@ -77,13 +79,13 @@ public TaskDefinitionResource create(String name, String definition) {
 		values.add("name", name);
 		values.add("definition", definition);
 		TaskDefinitionResource task = restTemplate.postForObject(
-				definitionsPath.expand(), values, TaskDefinitionResource.class);
+				definitionsPath.expand().getHref(), values, TaskDefinitionResource.class);
 		return task;
 	}
 
 	@Override
 	public void launch(String name, Map<String, String> properties) {
-		String uriTemplate = deploymentsPath.toString() + "/{name}";
+		String uriTemplate = deploymentsPath.getHref().toString() + "/{name}";
```

Comment:
should have grabbed the single resource link, and use it, rather than constructing URLs on the client side